walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,98 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //If there are no events or no one is attending the event, 
+    // all requests are possible if they are within the range of the whole day.
+    if(request.getDuration() <= TimeRange.WHOLE_DAY.duration()){
+      if(events.isEmpty() || request.getAttendees().isEmpty()) {
```

Comment:
minor nit here, but it feels cleaner to minimize nesting and separate out the different conditions (similar thing applies below as well). For example:
`if (duration > whole day) {
  return empty list
}
if (no events / attendees) {
  return whole day singleton
}
`

Reply:
Done